File: me_daily/2025/11/08/cf2152b/cf2152b.py
from collections import defaultdict
from typing import List
import sys
from collections import Counter
input = lambda: sys.stdin.readline().strip()
II = lambda: int(input())
LII = lambda: list(map(int, input().split()))
SI = lambda: input()
LSI = lambda: input().split()


t = II()


for _ in range(t):
    n, ra, ca, rb, cb = LII()

    ans = 0

    if rb > ra:
        ans = max(ans, rb)
    elif rb < ra:
        ans = max(ans, n - rb)
    if cb > ca:
        ans = max(ans, cb)
    elif cb < ca:
       ans = max(ans, n - cb)

    print(ans)

File: me_daily/2025/11/08/cf2152b/cf2152b.js
const readline = require('readline');
const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

// Token-based input (split by whitespace across lines)
const tokens = []; let idx = 0;
const next = () => tokens[idx++];
const nextInt = () => Number(next());
const nextStr = () => next();

// multiRead: pass parsers (e.g., Number, x=>x) and read multiple values
const multiRead = (...parsers) => parsers.map(p => p(next()));
// readArray: read n tokens with parser (default Number)
const readArray = (n, parser = Number) => Array.from({ length: n }, () => parser(next()));

const INF = 0x3f3f3f3f;
const MOD = 1000000007;

// Fast output helpers
const fastPrintArray = (arr, sep = ' ') => process.stdout.write(arr.join(sep) + '\n');
const fastPrintSlice = (slice, sep = ' ') => process.stdout.write(slice.join(sep) + '\n');

function solve() {
    const t = nextInt();
    const out = [];
    for (let _ = 0; _ < t; _++) {
        const [n, ra, ca, rb, cb] = multiRead(Number, Number, Number, Number, Number);
        let ansRow = rb > ra ? rb : (rb < ra ? (n -rb) : 0);
        let ansCol = cb > ca ? cb : (cb < ca ? (n - cb) : 0);
        const ans = Math.max(ansCol,ansRow);
        out.push(String(ans));
    }

    process.stdout.write(out.join('\n'));
}

rl.on('line', (line) => {
    if (line.length) tokens.push(...line.trim().split(/\s+/));
});
rl.on('close', () => { solve(); });
